code-challenges.test.js

```diff
@@ -20,9 +20,53 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+describe ("shuffledSubset", () => {
+    it("takes in an array, removes the first item from the array and shuffles the remaining items", () => {
+        expect(shuffledSubset(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+        expect(shuffledSubset(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+})
+
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// input: array
+// output: shuffled array without first value
+
+const shuffledSubset = (array) => {
+    // Use .shift to remove first value in the array
+    array.shift()
+
+    // This array will hold the randomly selected values
+    var shuffledArray = []
+    
+    // This array is for transferring values to and from the modified (shifted) array
+    var temporaryArray = []
+
+    // Shuffle remaining values in the modified array
+    for (let i = array.length; i > 0; i = i - 1) {
+        // Generate a random index from the array
+        let randomIndex = Math.floor(Math.random() * i)
+        // Add the randomly selected value to the shuffled array
+        shuffledArray.push(array[randomIndex])
+
+        // Transfer values from modified array to the temporary array
+        for (let i = array.length - 1; i > randomIndex; i = i - 1) {
+            temporaryArray.push(array[i])
+            array.pop()
+            }
+
+        // Remove the randomly selected value from the array so it will not be selected again
+        array.pop()
+
+        // Transfer the values from the temporary array back into the modified array
+        for (let i = temporaryArray.length - 1; i >= 0; i = i - 1) {
+            array.push(temporaryArray[i])
+            temporaryArray.pop()
+        }
+    }
+    return shuffledArray
+}
```

Comment:
Logic looks great on this problem!  If you are looking for a challenge, try refactoring this down to one for loop or even using a higher-order function.